packages/neuron-ui/.eslintrc.js

```diff
@@ -13,92 +11,62 @@ module.exports = {
       version: 'detect',
     },
   },
+  env: {
+    jest: true,
+    node: true,
+    browser: true,
+  },
+  globals: {
+    BigInt: 'readonly',
+  },
   rules: {
-    'prettier/prettier': [
-      2,
-      {
-        printWidth: 120,
-        endOfLine: 'auto',
-      },
-    ],
-    semi: [2, 'never'],
-    curly: [2, 'all'],
-    'comma-dangle': [
-      2,
-      {
-        arrays: 'always-multiline',
-        objects: 'always-multiline',
-        imports: 'always-multiline',
-        exports: 'always-multiline',
-        functions: 'ignore',
-      },
-    ],
-    'import/no-extraneous-dependencies': [
-      2,
-      {
-        devDependencies: true,
-      },
-    ],
-    'no-unused-vars': 'off',
-    'implicit-arrow-linebreak': 'off',
+    // TODO: Some temporarily disabled rules will be re-enabled later, considering that many files are affected and will be addressed in the future.
+    '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+
+    // This is the configuration that was set when using eslint-plugin-prettier
+    // https://github.com/prettier/eslint-plugin-prettier#arrow-body-style-and-prefer-arrow-callback-issue
+    'arrow-body-style': 'off',
+    'prefer-arrow-callback': 'off',
+
+    // https://github.com/jsx-eslint/eslint-plugin-react/issues/498
+    'react/prop-types': 'off',
+
+    // TypeScript support
     '@typescript-eslint/no-unused-vars': [
       'error',
       {
-        vars: 'local',
-        args: 'after-used',
-        ignoreRestSiblings: false,
+        argsIgnorePattern: '^_',
       },
     ],
-    'arrow-parens': [2, 'as-needed'],
-    'max-len': [
-      2,
-      {
-        code: 120,
-        ignoreComments: true,
-        ignoreTrailingComments: true,
-        ignoreUrls: true,
-        ignoreStrings: true,
-        ignoreTemplateLiterals: true,
-        ignoreRegExpLiterals: true,
-      },
-    ],
-    'object-curly-newline': [
+    // Avoid duplicating @typescript-eslint/no-shadow
+    'no-shadow': 'off',
+    '@typescript-eslint/no-shadow': 'error',
+
+    // Unnecessary rules
+    'no-plusplus': 'off',
+    'max-classes-per-file': 'off',
+    'no-use-before-define': 'off',
```

Comment:
I believe the benefits outweigh the costs, as many patterns require definitions to be placed later, such as a file that needs to highlight its main parts and put secondary parts at the bottom. This can improve readability.

Reply:
IMO, the improvement in readability introduced by ordering functions defined reversely is limited in static language because the compiler can resolve them correctly.

JavaScript, in dynamic languages, is special because it has `hoisting`, so the following code works

```javascript
print('hello')
function print (msg) {
  console.log(msg)
}
```

It works as a static language

But it's not actually recommended after `let`, `const` are introduced because the code becomes obscure when used-before-defined, it's hard to know if the code runs

```javascript
print('hello')
function print (msg) {
  console.log(msg)
}
```
works while

```javascript
print('hello')
const print = (msg) => {
  console.log(msg)
}
```
doesn't.

Reply:
If we disable this rule, the improvement in readability becomes evident, as in the following case:
```javascript
// bad
import something...

const var1 = ...
const var2 = ...

function fn1() { ... }
function fn2() { ... }

function SubComp() {
  ... code that relies on vars and fns ...
}

export function Comp() {
  ... code that relies on vars and fns and SubComp ...
}

// good
import something...

export function Comp() {
  ... code that relies on vars and fns and SubComp ...
}

function SubComp() {
  ... code that relies on vars and fns ...
}

// It can also be placed after the `import something`, and the position can be determined based on their significance.
const var1 = ...
const var2 = ...

function fn1() { ... }
function fn2() { ... }
```
This is a scenario for React components, and similar scenarios exist for class files, hook files, large utility function files, and more.
Their common characteristic is that the core content of the file is clearly defined. If we place the most important part at the beginning (followed by other parts in order of importance), it can significantly improve readability, because that is the content people are most interested in.
Just like when reading an article and first seeing the table of contents, it is helpful to have an overview of the core content and jump to specific sections if needed. For example, jumping from `Comp` to `SubComp` to focus on the desired information.

By enabling this rule, we lose the potential for improved readability, and the benefits it brings are limited to assisting us in statically checking simple scenarios that we can already determine ourselves.
Because we generally can handle the issue of defining order ourselves and considering the sacrifice it makes in terms of readability, I categorized it as an unnecessary rule.

Reply:
I think the disadvantage that brings by disabled the rule outweighs the advantage.

Reply:
My perspective is that most JavaScript programmers can handle variable hoisting order issues well (based on intuitive and quick judgments), and if any issues arise, they are easy to troubleshoot because this type of error provides clear guidance. Therefore, the cost of disabling this rule is minimal. 

However, based on the current difference of opinions, I will revert this rule back to the enabled state because the most important aspect in such matters is to establish a unified convention among project participants to enhance development efficiency, rather than focusing too much on debates about `lint` preferences.

Reply:
updated